patches/server/0355-Anti-Xray.patch

```diff
@@ -407,7 +410,7 @@ index 0000000000000000000000000000000000000000..3916d99c746d952e7ae6c3dde750d08f
 +                return switch (level.getWorld().getEnvironment()) {
 +                    case NETHER -> presetBlockStatesNetherrack;
 +                    case THE_END -> presetBlockStatesEndStone;
-+                    default -> presetBlockStatesStone;
++                    default -> bottomBlockY <= 4 ? presetBlockStatesDeepslate : presetBlockStatesStone; // Deepslate starts mixing with stone from y=8 and below. At y=0 all stone is deepslate, for an average of y=4.
```

Comment:
This won't as you might think. This method is called per chunk section. `bottomBlockY` can only be `..., -16, 0, 16, ...`.